src/main/java/entity/Goal.java

```diff
@@ -3,20 +3,21 @@
 import java.time.LocalDate;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * Represents a goal that tracks progress on a target task over a specific time period (week or month).
- * Each goal has associated metadata (info), a date range, frequency requirements, and current progress tracking.
- *
- * TODO: Will be implemented in Sophia's story #1 for weekly/monthly goal tracking
+ * It tracks completion dates toward achieving the goal frequency within the goal's date range.
+ * Todo: If task not found, simulate user creating new task (dont think it should be part of this)
+ * Todo: edit info
  */
 public class Goal {
-    private final Info info;
-    private final BeginAndDueDates beginAndDueDates;
+    private Info info;
+    private BeginAndDueDates beginAndDueDates;
     private final Task targetTask;
-    private final TimePeriod timePeriod;
-    private final int frequency;
-    private final List<LocalDate> completionDates;
+    private TimePeriod timePeriod;
+    private int frequency;
+    private List<LocalDate> completionDates;
```

Comment:
```suggestion
    private final TimePeriod timePeriod;
    private final int frequency;
    private final List<LocalDate> completionDates;
```
Entity fields should be ```final``` to enforce immutability. Only the contents of the `completionDates` list should be modifiable through specific domain methods like `recordCompletion()`.